discovery-infra/tests/base_test.py

```diff
@@ -40,39 +43,38 @@ def get_nodes(self) -> Callable:
         def get_nodes_func(config: Optional[TerraformConfig] = None):
             if not config:
                 config = TerraformConfig()
-            nodes_data["config"] = config
 
             if "nodes" in nodes_data:
                 return nodes_data["nodes"]
 
             net_asset = LibvirtNetworkAssets()
             config.net_asset = net_asset.get()
-            nodes_data["net_asset"] = net_asset
-
             controller = TerraformController(config)
             nodes = Nodes(controller, config.private_ssh_key_path)
-            nodes_data["nodes"] = nodes
 
             nodes.prepare_nodes()
             interfaces = BaseTest.nat_interfaces(config)
             nat = NatController(interfaces, NatController.get_namespace_index(interfaces[0]))
             nat.add_nat_rules()
-            nodes_data["nat"] = nat
 
+            nodes_data["nodes"] = nodes
+            nodes_data["config"] = config
+            nodes_data["net_asset"] = net_asset
+            nodes_data["nat"] = nat
             return nodes
 
         yield get_nodes_func
 
         _nodes: Nodes = nodes_data.get("nodes")
-        _config: TerraformConfig = nodes_data.get("config")
-        _nat: NatController = nodes_data.get("nat")
-        _net_asset: LibvirtNetworkAssets = nodes_data.get("net_asset")
+        _config: TerraformConfig = nodes_data["config"]
+        _nat: NatController = nodes_data["nat"]
+        _net_asset: LibvirtNetworkAssets = nodes_data["net_asset"]
 
         try:
-            if _nodes and global_variables.test_teardown:
+            if global_variables.test_teardown:
                 logging.info('--- TEARDOWN --- node controller\n')
-                _nodes.destroy_all_nodes()
-
+                if _nodes:
+                    _nodes.destroy_all_nodes()
```

Comment:
please take a look at this. Are those changes required or a bad rebase?

Reply:
I think you made the [changes](https://github.com/openshift/assisted-test-infra/pull/793/files#diff-6162d2a9234339ec7bb76b9037da27056716d9d64db599e223c8d132bc05787d) in this part. Should I restore?

Reply:
I'm pretty sure @eliorerz covered it on https://github.com/openshift/assisted-test-infra/pull/905 . I don't think the change I made is required anymore.

Reply:
I restored upstream version, Thanks!